### PR TITLE
[1227] Redirect to origin page when editing ethnic group/background

### PR DESCRIPTION
**Pull request by darokel**
### Context

- https://trello.com/c/2Q0wgbQg/1227-change-ethnicity-also-takes-the-user-through-the-disability-forms

### Changes proposed in this pull request

- Return to last origin page if they have come from the diversity confirm otherwise resume normal flow

fix

### Guidance to review

- Fill out a trainee's diversity details. Go back into the section to change ethnic group/background and submit. Assert you're taken back to the confirm page

### CODE REVIEW

**Comment by darokel** [comment]
[Change ethnicity also takes the user through the disability forms](https://trello.com/c/2Q0wgbQg/1227-change-ethnicity-also-takes-the-user-through-the-disability-forms)

File: app/controllers/trainees/diversity/ethnic_backgrounds_controller.rb [hunk 1]
```diff
@@ -41,6 +41,12 @@ def ethnic_background_params
       def background_is_different?(required_params)
         trainee.ethnic_background.present? && trainee.ethnic_background != required_params[:ethnic_background]
       end
+
+      def origin_page_or_next_step
+        return page_tracker.last_origin_page_path if page_tracker.last_origin_page_path&.include?("diversity/confirm")
```
[the remark this answers is not in the document]
Comment (darokel) [reply app/controllers/trainees/diversity/ethnic_backgrounds_controller.rb:46]: I personally wouldn't want to mix controller business logic into the page tracker so we keep it simple 🤷

**Comment by darokel** [comment]
> Looks good. Is it worth adding a test for this logic?

done https://github.com/DFE-Digital/register-trainee-teachers/pull/627/commits/8134c5fa25e70f44f9b70879139d333a51e22d34

**Review by gpeng — APPROVED** [review]

**Comment by gpeng** [comment]
I've approved so as we can get this in. I think we should try and avoid adding too many `context`-ses to the feature specs as they're slow. We can look at that in the code review we're going to do.

**Pull request merged**